### PR TITLE
Bug 2033720: Bump k8 dependencies to v0.23.1

### PR DESCRIPTION
**Pull request by dperaza4dustbit**
Signed-off-by: David Peraza <dperaza@redhat.com>

### CODE REVIEW

**Comment by gabemontero** [comment]
this will need https://github.com/openshift/release/pull/25443 for the compile to work in CI

File: go.mod [hunk 1]
```diff
@@ -14,8 +14,87 @@ require (
...
-	k8s.io/apimachinery v0.22.1
-	k8s.io/apiserver v0.22.1 // indirect
-	k8s.io/client-go v0.22.1
+	k8s.io/api v0.23.1
+	k8s.io/apimachinery v0.23.1
+	k8s.io/apiserver v0.23.1 // indirect
+	k8s.io/client-go v0.23.1
+)
+
+require (
```
Comment (gabemontero) [inline go.mod:23]: I tried the bump locally and did not need all this require stuff @dperaza4dustbit 

remove it and re-run `go mod tidy` then `go mod vendor`

if for some reason you can't do this on your Mac without this, then I'll submit an alternative PR

Comment (dperaza4dustbit) [reply go.mod:23]: Did you move go to 1.7 at the top of go.mod?

Comment (gabemontero) [reply go.mod:23]: yep your are right @dperaza4dustbit 

and now looking at it again, I see it now does 1 require blocks ... the top one with direct dependencies and the bottom one with indirect one 

let's leave as is and retest after merging the release PRs

**Comment by gabemontero** [comment]
ok master branch ocp is now golang 1.17 ... let's see 

/retest

**Comment by gabemontero** [comment]
still built with 1.16 ... may have retest too soon where the changes were not pushed everywhere needed 

let's wait a bit, then try again

**Comment by gabemontero** [comment]
@dperaza4dustbit if the images job fails again with it still using golang 1.16 we may have to change the Dockerfiles as well to get things to work (didn't think that was necessary but something might have changed)

**Comment by gabemontero** [comment]
> @dperaza4dustbit if the images job fails again with it still using golang 1.16 we may have to change the Dockerfiles as well to get things to work (didn't think that was necessary but something might have changed)

hmm did it again ... I'm confused, given the rehearsals passed  ... still may be a slow CI sync

that said, updating the Dockerfiles for local development is a good idea @dperaza4dustbit  ... go ahead and update them in a separate commit

File: Dockerfile.okd [hunk 2]
```diff
@@ -1,4 +1,4 @@
-FROM registry.svc.ci.openshift.org/openshift/release:golang-1.16 AS builder
+FROM registry.ci.openshift.org/openshift/release:golang-1.17 AS builder
```
Comment (gabemontero) [inline Dockerfile.okd:1]: you need to change line 6 below too

FROM registry.ci.openshift.org/origin/4.10:base

**Comment by gabemontero** [comment]
PR e2e's are sufficient testing

/label qe-approved

File: Dockerfile.okd (hunk 2 above)
Comment (gabemontero) [reply Dockerfile.okd:1]: if the e2e's stay green I'll handle this in a separate PR

**Comment by gabemontero** [comment]
/hold

@dperaza4dustbit and I are investigating

looks like the unit test still wants to run golang 1.16 for some reason

**Comment by yselkowitz** [comment]
Note that #403 already exists to move to go1.17

**Comment by gabemontero** [comment]
AWS fail

/test e2e-aws-image-ecosystem

**Pull request merged**